pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -147,13 +147,21 @@ func IsNodeZero(n *yaml.Node) bool {
 		n.Line == 0 && n.Column == 0
 }
 
-// validatePath validates input path and return an error if it's invalid
-func validatePath(p string) error {
+// validateFnConfigPath validates given function config path and return an error if it's invalid
+func validateFnConfigPath(p string) error {
+	if strings.TrimSpace(p) == "" {
+		return fmt.Errorf("path must not be empty")
+	}
+	p = path.Clean(p)
 	if path.IsAbs(p) {
-		return fmt.Errorf("path is not relative")
+		return fmt.Errorf("path must be relative")
 	}
-	if strings.TrimSpace(p) == "" {
-		return fmt.Errorf("path cannot have only white spaces")
+	if strings.HasPrefix(p, "..") {
+		// fn config must not live outside the package directory
+		// Allowing outside path opens up an attack vector that allows
+		// reading any YAML file on package consumer's machine.
+		return fmt.Errorf("path must not be outside the package directory")
+
 	}
 	if p != sourceAllSubPkgs && strings.Contains(p, "*") {
 		return fmt.Errorf("path contains asterisk, asterisk is only allowed in './*'")
```

Comment:
what's `./*` in the context of `configPath`?

Reply:
Removed the `./*` bit. This code was originally for validating `pipeline.Sources` which we took it out. Cleaned it up.